File: aryn-mall-ui/apps/web-ele/src/api/promotion/distribution-user.ts
import { requestClient } from '#/api/request';

export interface DistributionUserRecord {
  availableCommission: number;
  avatar?: string;
  commissionDebt: number;
  createTime?: string;
  frozenCommission: number;
  id: string;
  inviterUserId?: string;
  nickname?: string;
  pendingCommission: number;
  status: '0' | '1';
  subordinateCount: number;
  totalCommission: number;
  userId: string;
  withdrawnCommission: number;
}

export interface DistributionUserQuery {
  current: number;
  inviterUserId?: string;
  size: number;
  userId?: string;
}

export interface DistributionUserPage {
  current: number;
  records: DistributionUserRecord[];
  size: number;
  total: number;
}

export interface DistributionUserRegisterPayload {
  avatar?: string;
  inviterUserId?: string;
  nickname?: string;
  userId: string;
}

/**
 * 获取分销用户分页
 */
export async function getPage(query: DistributionUserQuery) {
  return requestClient.get<DistributionUserPage>(
    '/promotion/distribution/user/page',
    {
      params: query,
    },
  );
}

/**
 * 根据ID获取分销用户
 */
export async function getById(id: string) {
  return requestClient.get<DistributionUserRecord>(
    `/promotion/distribution/user/${id}`,
  );
}

/**
 * 注册分销用户
 */
export async function register(data: DistributionUserRegisterPayload) {
  return requestClient.post<DistributionUserRecord>(
    '/promotion/distribution/user/register',
    data,
  );
}

/**
 * 启用分销用户
 */
export async function enable(userId: string) {
  return requestClient.put<boolean>(
    `/promotion/distribution/user/enable/${userId}`,
  );
}

/**
 * 禁用分销用户
 */
export async function disable(userId: string) {
  return requestClient.put<boolean>(
    `/promotion/distribution/user/disable/${userId}`,
  );
}

/**
 * 删除分销用户
 */
export async function delObj(id: string) {
  return requestClient.delete<boolean>(`/promotion/distribution/user/${id}`);
}

File: aryn-mall-java/aryn-promotion/aryn-promotion-api/src/main/java/com/aryn/cloud/promotion/api/entity/DistributionUser.java
package com.aryn.cloud.promotion.api.entity;

import com.baomidou.mybatisplus.annotation.*;
import com.baomidou.mybatisplus.extension.activerecord.Model;
import io.swagger.v3.oas.annotations.media.Schema;
import jakarta.validation.constraints.NotBlank;
import lombok.Data;
import lombok.EqualsAndHashCode;
import lombok.experimental.Accessors;

import java.math.BigDecimal;
import java.time.LocalDateTime;

/**
 * 分销用户
 *
 * @author 雨滴kian
 * @date 2025/4/8
 */
@Data
@Schema(description = "分销用户")
@EqualsAndHashCode(callSuper = true)
@Accessors(chain = true)
@TableName("distribution_user")
public class DistributionUser extends Model<DistributionUser> {

	@Schema(description = "主键")
	@TableId(type = IdType.ASSIGN_ID)
	private String id;

	@Schema(description = "用户ID")
	@NotBlank(message = "用户ID不能为空")
	private String userId;

	@Schema(description = "用户昵称")
	private String nickname;

	@Schema(description = "用户头像")
	private String avatar;

	@Schema(description = "邀请人ID")
	private String inviterUserId;

	@Schema(description = "累计佣金")
	private BigDecimal totalCommission;

	@Schema(description = "可提现佣金")
	private BigDecimal availableCommission;

	@Schema(description = "待结算佣金")
	private BigDecimal pendingCommission;

	@Schema(description = "已提现佣金")
	private BigDecimal withdrawnCommission;

	@Schema(description = "冻结佣金（提现申请中）")
	private BigDecimal frozenCommission;

	@Schema(description = "退款产生的佣金欠款")
	private BigDecimal commissionDebt;

	@Schema(description = "下级人数")
	private Integer subordinateCount;

	@Schema(description = "状态：0启用 1禁用")
	private String status;

	@TableField(fill = FieldFill.INSERT)
	@Schema(description = "创建人")
	private String createBy;

	@TableField(fill = FieldFill.UPDATE)
	@Schema(description = "修改人")
	private String updateBy;

	@TableField(fill = FieldFill.INSERT)
	@Schema(description = "创建时间")
	private LocalDateTime createTime;

	@TableField(fill = FieldFill.UPDATE)
	@Schema(description = "修改时间")
	private LocalDateTime updateTime;

	@TableLogic
	@TableField(fill = FieldFill.INSERT)
	@Schema(description = "逻辑删除：0显示 1隐藏")
	private String delFlag;

	@Schema(description = "租户id")
	private String tenantId;

	@Schema(description = "版本号")
	@Version
	private Integer version;

}
